Python/libraries/recognizers-date-time/recognizers_date_time/date_time/base_datetimeperiod.py

```diff
@@ -312,88 +312,104 @@ def is_valid_connector_for_date_and_time_period(self, text: str):
+
+            if not match:
+                match = regex.search(self.config.am_desc_regex, after_str)
+
+            if not match or after_str[0:match.start()]:
+                match = regex.search(self.config.pm_desc_regex, after_str)
 
             if match:
-                if not before_str[match.end():].strip():
-                    middle_str = source[match.end():extracted_result.start]
-                    if middle_str == ' ':
-                        tokens.append(
-                            Token(match.start(), extracted_result.start + extracted_result.length))
+                if not after_str[0:match.start()]:
+                    tokens.append(Token(extracted_result.start, extracted_result.end + match.end()))
+
+            prefix_str = source[0: extracted_result.start]
+
+            match = regex.search(self.config.period_time_of_day_with_date_regex, prefix_str)
+            if match:
+                if not prefix_str[match.end():] or prefix_str[match.end():].isspace():
+                    mid_str = source[match.end(): extracted_result.start]
+                    if mid_str and mid_str.isspace():
+                        tokens.append(Token(match.start(), extracted_result.start + extracted_result.length))
+                        m_start = match.start()
                 else:
-                    pause_match = regex.search(
-                        self.config.middle_pause_regex, before_str[match.end():])
-                    if pause_match:
-                        suffix = source[extracted_result.start + extracted_result.length:].strip()
+                    connector_str = prefix_str[match.end():]
 
-                        ending_match = regex.search(
-                            self.config.general_ending_regex, suffix)
+                    # Trim here is set to false as the Regex might catch white spaces before or after the text
+                    if RegexExtension.is_exact_match(self.config.middle_pause_regex, connector_str, False):
+                        suffix = source[extracted_result.start + extracted_result.length:].strip(' ')
+                        ending_match = self.config.general_ending_regex.match(suffix)
                         if ending_match:
-                            tokens.append(
-                                Token(match.start(), extracted_result.start + extracted_result.length))
+                            tokens.append(Token(match.start(), extracted_result.start + extracted_result.length))
 
-            # check whether there are adjacent time period strings, before or after
-            for token in tokens:
-                # try to extract a time period in before-string
-                if token.start > 0:
-                    before_str = source[0:token.start].strip()
-                    if before_str:
-                        ers_time = self.config.time_period_extractor.extract(
-                            before_str, reference)
+        result = list(tokens)
```

Comment:
Same here about change to `[]`

Reply:
In this case it has to be this way because "result" is being matched to a new instance of the "tokens" list, so that there is no conflict with the original when doing the "for-in" loop later. Regarding the other one, I'll change them now, thank you!